components/director/internal/domain/application/service.go

```diff
@@ -1238,6 +1239,49 @@ func (s *service) genericUpsert(ctx context.Context, appTenant string, in model.
 		return errors.Wrapf(err, "while creating multiple labels for Application with id %s", id)
 	}
 
+	if err = s.createWebhooksIfNotExist(ctx, app.ID, appTenant, in.Webhooks); err != nil {
+		return errors.Wrapf(err, "while processing webhooks for application with id %q", app.ID)
+	}
+
+	return nil
+}
+
+func (s *service) createWebhooksIfNotExist(ctx context.Context, appID, appTenant string, appWebhooks []*model.WebhookInput) error {
+	if len(appWebhooks) == 0 {
+		return nil
+	}
+
+	webhooksFromDB, err := s.webhookRepo.ListByReferenceObjectID(ctx, appTenant, appID, model.ApplicationWebhookReference)
+	if err != nil {
+		return errors.Wrapf(err, "while listig webhooks for application with id %q", appID)
+	}
+
+	webhooks := make([]*model.Webhook, 0, len(appWebhooks))
+	for _, item := range appWebhooks {
+		webhooks = append(webhooks, item.ToWebhook(s.uidService.Generate(), appID, model.ApplicationWebhookReference))
+	}
+
+	webhooksToCreate := make([]*model.Webhook, 0, len(appWebhooks))
+	for _, wh := range webhooks {
+		found := false
+		for _, whDB := range webhooksFromDB {
+			if wh.Type == whDB.Type && wh.ObjectID == whDB.ObjectID && str.PtrStrToStr(wh.URL) == str.PtrStrToStr(whDB.URL) {
```

Comment:
What's the point of `wh.ObjectID == whDB.ObjectID` being in the conditional?

Reply:
Shouldn't the above `s.webhookRepo.ListByReferenceObjectID` guarantee that the webhooksFromDb will have this particular ObjectID?

Reply:
honestly this check has been added for readability reasons.